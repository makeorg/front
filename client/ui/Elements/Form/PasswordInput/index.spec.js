import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { PasswordInput } from './index';

jest.mock('./Button', () => ({
  PasswordButton: 'PasswordButton',
}));

jest.mock('Client/ui/Elements/Form/Styled/Input', () => ({
  BasicInputStyle: 'BasicInputStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Content', () => ({
  MiddleFakeFieldStyle: 'MiddleFakeFieldStyle',
  FloatingLabelStyle: 'FloatingLabelStyle',
  FieldWrapperStyle: 'FieldWrapperStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Icons', () => ({
  CenterInputIconStyle: 'CenterInputIconStyle',
}));

describe('PasswordInput', () => {
  const type = 'foo';
  const name = 'bar';
  const icon = 'SvgLock';
  const value = 'baz';
  const label = 'qux';
  const handleChange = () => {};
  const togglePasswordIsDisplayed = () => {};

  it('must match the snapshot with default Props', () => {
    const component = renderer
      .create(
        <PasswordInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          togglePasswordIsDisplayed={togglePasswordIsDisplayed}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it('must return the diff between snapshot when password is displayed vs hidden', () => {
    const ShowPassword = renderer
      .create(
        <PasswordInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          togglePasswordIsDisplayed={togglePasswordIsDisplayed}
          passwordIsDisplayed
        />
      )
      .toJSON();
    const HidePassword = renderer
      .create(
        <PasswordInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          togglePasswordIsDisplayed={togglePasswordIsDisplayed}
          passwordIsDisplayed={false}
        />
      )
      .toJSON();
    expect(snapshotDiff(ShowPassword, HidePassword)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when password is required or optionnal', () => {
    const RequiredPassword = renderer
      .create(
        <PasswordInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          togglePasswordIsDisplayed={togglePasswordIsDisplayed}
          required
        />
      )
      .toJSON();
    const OptionnalPassword = renderer
      .create(
        <PasswordInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          togglePasswordIsDisplayed={togglePasswordIsDisplayed}
        />
      )
      .toJSON();
    expect(snapshotDiff(RequiredPassword, OptionnalPassword)).toMatchSnapshot();
  });
});
