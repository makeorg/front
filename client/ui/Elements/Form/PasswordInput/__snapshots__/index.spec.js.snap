// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput must match the snapshot with default Props 1`] = `
<MiddleFakeFieldStyle
  hasError={false}
>
  <CenterInputIconStyle>
    SvgLock
  </CenterInputIconStyle>
  <FieldWrapperStyle>
    <BasicInputStyle
      aria-invalid={false}
      id="bar"
      minLength={8}
      name="bar"
      onChange={[Function]}
      required={true}
      type="password"
      value="baz"
    />
    <FloatingLabelStyle
      htmlFor="bar"
    >
      qux
    </FloatingLabelStyle>
  </FieldWrapperStyle>
  <PasswordButton
    isPasswordDisplayed={false}
    toggleIsPasswordDisplayed={[Function]}
  />
</MiddleFakeFieldStyle>
`;

exports[`PasswordInput must return the diff between snapshot when password is displayed vs hidden 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`PasswordInput must return the diff between snapshot when password is required or optionnal 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
